C/jets-secp256k1.c

```diff
@@ -0,0 +1,607 @@
+  read_scalar(&a, &src);
+  secp256k1_scalar_negate(&a, &a);
+  write_scalar(dst, &a);
+  return true;
+}
+
+bool scalar_add(frameItem* dst, frameItem src, const txEnv* env) {
+  (void) env; // env is unused;
+
+  secp256k1_scalar a, b;
+  read_scalar(&a, &src);
+  read_scalar(&b, &src);
+  secp256k1_scalar_add(&a, &a, &b);
+  write_scalar(dst, &a);
+  return true;
+}
+
+bool scalar_square(frameItem* dst, frameItem src, const txEnv* env) {
+  (void) env; // env is unused;
+
+  secp256k1_scalar a;
+  read_scalar(&a, &src);
+  secp256k1_scalar_mul(&a, &a, &a);
+  write_scalar(dst, &a);
+  return true;
+}
+
+bool scalar_multiply(frameItem* dst, frameItem src, const txEnv* env) {
+  (void) env; // env is unused;
+
+  secp256k1_scalar a, b;
+  read_scalar(&a, &src);
+  read_scalar(&b, &src);
+  secp256k1_scalar_mul(&a, &a, &b);
+  write_scalar(dst, &a);
+  return true;
+}
+
+bool scalar_multiply_lambda(frameItem* dst, frameItem src, const txEnv* env) {
+  (void) env; // env is unused;
+
+  secp256k1_scalar a;
+  read_scalar(&a, &src);
+  secp256k1_scalar_mul(&a, &a, &secp256k1_const_lambda);
+  write_scalar(dst, &a);
+  return true;
+}
+
+bool scalar_invert(frameItem* dst, frameItem src, const txEnv* env) {
+  (void) env; // env is unused;
+
+  secp256k1_scalar a;
+  read_scalar(&a, &src);
+  secp256k1_scalar_inverse_var(&a, &a);
+  write_scalar(dst, &a);
+  return true;
+}
+
+bool scalar_is_zero(frameItem* dst, frameItem src, const txEnv* env) {
+  (void) env; // env is unused;
+
+  secp256k1_scalar a;
+  read_scalar(&a, &src);
+  writeBit(dst, secp256k1_scalar_is_zero(&a));
+  return true;
+}
+
+bool gej_infinity(frameItem* dst, frameItem src, const txEnv* env) {
+  (void) src; // src is unused;
+  (void) env; // env is unused;
+
+  secp256k1_gej a;
+  secp256k1_gej_set_infinity(&a);
+  write_gej(dst, &a);
+  return true;
+}
+
+bool gej_rescale(frameItem* dst, frameItem src, const txEnv* env) {
+  (void) env; // env is unused;
+
+  secp256k1_gej a;
+  secp256k1_fe c;
+  read_gej(&a, &src);
+  read_fe(&c, &src);
+  secp256k1_gej_rescale(&a, &c);
+  write_gej(dst, &a);
+  return true;
+}
+
+bool gej_normalize(frameItem* dst, frameItem src, const txEnv* env) {
+  (void) env; // env is unused;
+
+  secp256k1_gej a;
+  secp256k1_ge r;
+  read_gej(&a, &src);
+  secp256k1_ge_set_gej_var(&r, &a);
+  write_ge(dst, &r);
+  return true;
+}
+
+bool gej_negate(frameItem* dst, frameItem src, const txEnv* env) {
+  (void) env; // env is unused;
+
+  secp256k1_gej a;
+  read_gej(&a, &src);
+  secp256k1_gej_neg(&a, &a);
+  write_gej(dst, &a);
+  return true;
+}
+
+bool ge_negate(frameItem* dst, frameItem src, const txEnv* env) {
+  (void) env; // env is unused;
+
+  secp256k1_ge a;
+  read_ge(&a, &src);
+  secp256k1_ge_neg(&a, &a);
+  write_ge(dst, &a);
+  return true;
+}
+
+bool gej_double(frameItem* dst, frameItem src, const txEnv* env) {
+  (void) env; // env is unused;
+
+  // This r variable can be eliminated once secp256k1_gej_double_var calls set_infinity.
```

Comment:
This comment suggests that the code should have been updated in 19d3074911dc56c4f06239c700cd76cddc8517ec, when `secp256k1_gej_double_var` is changed to set the infinity flag. (Though note that it does **not** call `set_infinity` and therefore may not zero out the x and y coordinates.)

Reply:
Fixed.